### PR TITLE
feat: add style prop in Modal

### PR DESCRIPTION
**Pull request by ahce**
### Motivation

This prop is necessary when you need to see content in full screen on iPhone X/11

### Test plan
1. Download this branch
2. Open Modal example
3. Pass style prop with this values (iPhone X/11):
```js
 { marginTop: 0, marginBottom: 0 }
```
### Actual behaviour
image

### Behaviour with  style = { marginTop: 0, marginBottom: 0 }
image

### CODE REVIEW

File: src/components/Modal.tsx [hunk 1]
```diff
@@ -17,6 +17,10 @@ import Surface from './Surface';
 import { withTheme } from '../core/theming';
 
 type Props = {
+  /**
+   * Determine whether the insets are applied.
```
Comment (Trancever) [inline src/components/Modal.tsx:21]:
```suggestion
   * Determines whether the insets are applied.
```

**Review by Trancever — CHANGES_REQUESTED** [review]

File: src/components/Modal.tsx [hunk 2]
```diff
@@ -17,6 +17,10 @@ import Surface from './Surface';
...
+   */
+  disableSafeAreaInsets?: boolean;
```
Comment (Trancever) [inline src/components/Modal.tsx:23]: We would prefer to have `safeAreaInsets` prop instead where users could pass `top` and `bottom` properties to override the default behavior. Could you change the implementation, please?

**Review by Trancever — CHANGES_REQUESTED** [review]

File: src/components/Modal.tsx (hunk 2 above)
Comment (ahce-meli) [reply src/components/Modal.tsx:23]: @Trancever you think is better pass a wrapperStyle?

Comment (ahce) [reply src/components/Modal.tsx:23]: you think is better pass a wrapperStyle?

Comment (Trancever) [reply src/components/Modal.tsx:23]: Yes, we can even call it simply `style` :) Please provide a nice description for this prop so it's clear what can be achieved using it. Thanks!

Comment (ahce) [reply src/components/Modal.tsx]: done!

**Review by Trancever — APPROVED** [review]

**Pull request merged**